watertap/property_models/activated_sludge/asm1_properties.py

```diff
@@ -282,39 +283,76 @@ def build(self):
             units=pyo.units.kmol / pyo.units.m**3,
         )
 
-    def get_material_flow_terms(b, p, j):
-        if j == "H2O":
-            return b.flow_vol * b.params.dens_mass
-        elif j == "S_ALK":
-            # Convert moles of alkalinity to mass of C assuming all is HCO3-
-            return b.flow_vol * b.alkalinity * (12 * pyo.units.kg / pyo.units.kmol)
-        else:
-            return b.flow_vol * b.conc_mass_comp[j]
-
-    def get_enthalpy_flow_terms(b, p):
-        return (
-            b.flow_vol
-            * b.params.dens_mass
-            * b.params.cp_mass
-            * (b.temperature - b.params.temperature_ref)
+        # Material and energy flow and density expressions
+        def material_flow_expression(self, j):
+            if j == "H2O":
+                return self.flow_vol * self.params.dens_mass
+            elif j == "S_ALK":
+                # Convert moles of alkalinity to mass of C assuming all is HCO3-
+                return (
+                    self.flow_vol
+                    * self.alkalinity
+                    * (12 * pyo.units.kg / pyo.units.kmol)
```

Comment:
Is this `12 * pyo.units.kg / pyo.units.kmol` the molar mass of the solute? Then why not set it as a variable?

Reply:
No, this is the molecular weight of carbon, and is a constant which will not change.

Reply:
A minor point, I think keep this is OK, but maybe add it as a parameter can be another option.